lib/http/request.js

```diff
@@ -1,37 +1,47 @@
 /**
- * Module dependencies.
+ * @file Module dependencies.
  */
 'use strict';
 
-// const http = require('http')
-//  , req = http.IncomingMessage.prototype;
-
 /* eslint-disable no-multi-assign */
+/**
+ * @namespace {PlainObject} HttpRequest
+ */
 const req = exports = module.exports = {};
 /* eslint-enable no-multi-assign */
 
 /**
-* @typedef {Object} LogInOptions
-* @property {boolean} [session] Save login state in session, defaults to _true_
+* @callback LoginDoneCallback
+* @param {Error} [error]
+*/
+
+/**
+* @typedef {PlainObject} LogInOptions
+* @property {boolean} [session=true] Save login state in session.
 */
 
 /**
  * Initiate a login session for `user`.
  *
- *
+ * @function HttpRequest#logIn
  * @example
  *
  *     req.logIn(user, { session: false });
  *
- *     req.logIn(user, (err) => {
- *       if (err) { throw err; }
- *       // session saved
- *     });
+ *     (async () => {
+ *     try {
+ *      await req.logIn(user);
+ *     } catch (err) {
+ *      console.error(err);
```

Comment:
Remove the console.error we shouldn't be logging to console

Reply:
That's just part of the example to drive home the point that the `await` can be caught (in a way which avoids the linter complaining that there is a useless rethrow, as the linter also checks jsdoc examples). But I can disable the linter for that and avoid the console reporting in the example if you wish, or avoid the entire try-catch around the example if you like...

Reply:
Ignore that then, I didn't notice it was in the js doc!